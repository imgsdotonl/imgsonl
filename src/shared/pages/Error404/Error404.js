/* @flow */

import React from 'react';
import Helmet from 'react-helmet';
import PageTemplate from '../../theme/Imgsonl';

function Error404() {
  return (
    <div className="error_wrapper">
    <PageTemplate helmetMeta={ <Helmet title="404! You might be lost..." /> }>
      <p>Sorry, that page was not found.</p>
    </PageTemplate>
    </div>
  );
}

export default Error404;
