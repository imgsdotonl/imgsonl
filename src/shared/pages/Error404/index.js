/* @flow */

export { default } from './Error404';
