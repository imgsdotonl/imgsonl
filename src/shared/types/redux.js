/* @flow */

export type ApiAction<T> = {
  key: string,
  payload: T,
};

export type Store = {
  dispatch(action: ActionType): ActionType
}
export type ActionType =
  | 'LOGIN_REQUEST'
  | 'LOGIN_SUCCESS'
  | 'LOGIN_FAILURE'
  | 'LOGOUT_USER'
  | 'LOGOUT_USER_FAIL'
  | 'CHECK_AUTH_REQUEST'
  | 'CHECK_AUTH_SUCCESS'
  | 'CHECK_AUTH_FAILURE'
  | 'SIGNUP_USER_REQUEST'
  | 'SIGNUP_USER_SUCCESS'
  | 'SIGNUP_USER_FAILURE'
  | 'FORGOT_PASSWORD_REQUEST'
  | 'FORGOT_PASSWORD_SUCCESS'
  | 'FORGOT_PASSWORD_FAILURE'
  | 'RESET_PASSWORD_REQUEST'
  | 'RESET_PASSWORD_SUCCESS'
  | 'RESET_PASSWORD_FAILURE'
  | 'LOAD_POSTS_REQUEST'
  | 'LOAD_POSTS_SUCCESS'
  | 'LOAD_POSTS_FAILURE'
  | 'GET_POST_REQUEST'
  | 'GET_POST_SUCCESS'
  | 'GET_POST_FAILURE'
  | 'UPDATE_POST_REQUEST'
  | 'UPDATE_POST_SUCCESS'
  | 'UPDATE_POST_FAILURE'
  | 'SELECT_POST'
  | 'SELECT_POST_SUCCESS'
  | 'SELECT_POST_FAILURE'
  | 'CREATE_POST_REQUEST'
  | 'CREATE_POST_SUCCESS'
  | 'CREATE_POST_FAILURE'
  | 'DELETE_POST_FAILURE'
  | 'DELETE_POST_REQUEST'
  | 'DELETE_POST_SUCCESS';
