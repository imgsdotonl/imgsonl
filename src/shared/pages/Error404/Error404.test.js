import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Error404 from './Error404';

describe('<Error404 />', () => {
  const wrapper = shallow(<Error404 />);
  it('renders <Error404 /> without breaking', () => {
    expect(wrapper.find('div').length).toBe(1);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
