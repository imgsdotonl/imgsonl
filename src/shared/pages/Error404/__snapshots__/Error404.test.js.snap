exports[`<Error404 /> renders snapshot 1`] = `
<div
  className="error_wrapper">
  <Connect(Boldr)
    helmetMeta={
      <HelmetWrapper
        title="404! You might be lost..." />
    }>
    <p>
      Sorry, that page was not found.
    </p>
  </Connect(Boldr)>
</div>
`;
